test/npm/npm.spec.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+let Promise = require('bluebird');
+let expect = require('chai').expect;
+let http = require('http');
+let logger = require('mocha-logger');
+let retry = require('bluebird-retry');
+let spawn = require('child_process').spawn;
+
+function spawnify(command, args) {
+    return new Promise((resolve, reject) => {
+        logger.log('spawning "' + command + ' ' + args.join(' '));
+        var options = {
+            detached: true,
+            stdio: ['ignore', 'pipe', 'pipe']
+        }
+        var spawned = spawn(command, args, options);
+        var stdout = '';
+        var stderr = '';
+
+        spawned.stdout.on('data', (data) => {
+            logger.log('STDOUT:', data);
+            stdout += data;
+        });
+
+        spawned.stderr.on('data', (data) => {
+            logger.log('STDERR:', data);
+            stderr += data;
+        });
+
+        spawned.on('close', (code) => {
+            if (code === 0) {
+                resolve(stdout, stderr);
+            } else {
+                reject(Error('command: "' + command + ' ' + args.join(' ') +
+                             '", failed with ' + code));
+            }
+        });
+    });
+}
+
+function dockerBash(command) {
+    return spawnify('docker',
```

Comment:
docker commands do not require sudo if you follow the last few instructions that docker tells you after each install: 

```If you would like to use Docker as a non-root user, you should now consider
	adding your user to the "docker" group with something like:

	  sudo usermod -aG docker $your_user

	Remember that you will have to log out and back in for this to take effect!
```
Which is the right way to run docker.